investigation-service/src/main/java/gov/cdc/etldatapipeline/investigation/util/ProcessInvestigationDataUtil.java

```diff
@@ -45,235 +47,221 @@ public class ProcessInvestigationDataUtil {
+
+            investigationConfirmation.setConfirmationMethodTime(
+                    confirmationMethodTime == null ? phcLastChgTime : confirmationMethodTime);
+
+            for(Map.Entry<String, String> entry : confirmationMethodMap.entrySet()) {
+                investigationConfirmation.setConfirmationMethodCd(entry.getKey());
+                investigationConfirmation.setConfirmationMethodDescTxt(entry.getValue());
+                investigationConfirmationMethodKey.setConfirmationMethodCd(entry.getKey());
+                String jsonKey = jsonGenerator.generateStringJson(investigationConfirmationMethodKey);
+                String jsonValue = jsonGenerator.generateStringJson(investigationConfirmation);
+                kafkaTemplate.send(investigationConfirmationOutputTopicName, jsonKey, jsonValue);
             }
+        } catch (IllegalArgumentException ex) {
+            logger.info(ex.getMessage(), "InvestigationObservationIds");
         } catch (Exception e) {
             logger.error("Error processing investigation confirmation method JSON array from investigation data: {}", e.getMessage());
         }
     }
 
-    private void processInvestigationPageCaseAnswer(String investigationCaseAnswer, InvestigationTransformed investigationTransformed, ObjectMapper objectMapper) {
+    private void processInvestigationPageCaseAnswer(String investigationCaseAnswer, InvestigationTransformed investigationTransformed) {
         try {
-            JsonNode investigationCaseAnswerJsonArray = parseJsonArray(investigationCaseAnswer, objectMapper);
+            JsonNode investigationCaseAnswerJsonArray = parseJsonArray(investigationCaseAnswer);
 
-            if(investigationCaseAnswerJsonArray != null) {
-                Long actUid = investigationCaseAnswerJsonArray.get(0).get("act_uid").asLong();
-                List<InvestigationCaseAnswer> investigationCaseAnswerList = new ArrayList<>();
+            Long actUid = investigationCaseAnswerJsonArray.get(0).get("act_uid").asLong();
+            List<InvestigationCaseAnswer> investigationCaseAnswerList = new ArrayList<>();
 
-                for(JsonNode node : investigationCaseAnswerJsonArray) {
-                    InvestigationCaseAnswer tempCaseAnswerObject = objectMapper.treeToValue(node, InvestigationCaseAnswer.class);
-                    investigationCaseAnswerList.add(tempCaseAnswerObject);
-                }
+            for(JsonNode node : investigationCaseAnswerJsonArray) {
+                InvestigationCaseAnswer tempCaseAnswerObject = objectMapper.treeToValue(node, InvestigationCaseAnswer.class);
+                investigationCaseAnswerList.add(tempCaseAnswerObject);
+            }
 
-                investigationCaseAnswerRepository.deleteByActUid(actUid);
-                investigationCaseAnswerRepository.saveAll(investigationCaseAnswerList);
+            investigationCaseAnswerRepository.deleteByActUid(actUid);
```

Comment:
This will be phased out?

Reply:
The whole logic will be changed. Now it directly writes into rdb, new logic will just send a topic(s).